pkg/minishift/addon/addon_meta.go

```diff
@@ -31,13 +31,15 @@ const (
 	DescriptionMetaTagName   = "Description"
 	RequiredOpenShiftVersion = "OpenShift-Version"
 	anyOpenShiftVersion      = ""
+	varDefaults              = "Var-Defaults"
 )
 
 // AddOnMeta defines a set of meta data for an AddOn. Name and Description are required. Others are optional.
 type AddOnMeta interface {
 	Name() string
 	Description() []string
-	RequiredVars() []string
+	RequiredVars() ([]string, error)
+	VarDefaults() ([]string, error)
```

Comment:
A slice of strings? Are we are not talking key/value pairs here? I expect to get a map map[string]string or a slice of a struct RequiredVar (with two elements)

Reply:
IMO it is the metadata manager who should do the heavy lifting in terms of parsing the giving access to the metadata. This will also allow failing early in case there are syntactical issues in the metadata.